core/src/main/scala/org/apache/spark/status/api/v1/JacksonMessageWriter.scala

```diff
@@ -47,7 +47,6 @@ private[v1] class JacksonMessageWriter extends MessageBodyWriter[Object]{
       super.writeValueAsString(t)
     }
   }
-  mapper.registerModule(com.fasterxml.jackson.module.scala.DefaultScalaModule)
```

Comment:
This was added by @squito in #5940 in order to allow SparkStatusAPI POJOs to be serialized using Jackson. These POJOs are defined in https://github.com/apache/spark/blob/a4ead6d3881f071a2ae53ff1c961c6ac388cac1d/core/src/main/scala/org/apache/spark/status/api/v1/api.scala

I wonder if the default values in the `ApplicationAttemptInfo` are going to be handled differently without the `jackson-module-scala` stuff. We might have to explicitly add annotations in order to pin those defaults.